Modules/posixmodule.c

```diff
@@ -14334,44 +14340,96 @@ os_get_terminal_size_impl(PyObject *module, int fd)
 /*[clinic input]
 os.cpu_count
 
-Return the number of CPUs in the system; return None if indeterminable.
+Return the number of logical CPUs in the system.
 
-This number is not equivalent to the number of CPUs the current process can
-use.  The number of usable CPUs can be obtained with
-``len(os.sched_getaffinity(0))``
+Return None if indeterminable.
 [clinic start generated code]*/
 
 static PyObject *
 os_cpu_count_impl(PyObject *module)
-/*[clinic end generated code: output=5fc29463c3936a9c input=e7c8f4ba6dbbadd3]*/
+/*[clinic end generated code: output=5fc29463c3936a9c input=ba2f6f8980a0e2eb]*/
 {
-    int ncpu = 0;
+    int ncpu;
 #ifdef MS_WINDOWS
-#ifdef MS_WINDOWS_DESKTOP
+# ifdef MS_WINDOWS_DESKTOP
     ncpu = GetActiveProcessorCount(ALL_PROCESSOR_GROUPS);
-#endif
+# else
+    ncpu = 0;
+# endif
+
 #elif defined(__hpux)
     ncpu = mpctl(MPC_GETNUMSPUS, NULL, NULL);
+
 #elif defined(HAVE_SYSCONF) && defined(_SC_NPROCESSORS_ONLN)
     ncpu = sysconf(_SC_NPROCESSORS_ONLN);
+
 #elif defined(__VXWORKS__)
     ncpu = _Py_popcount32(vxCpuEnabledGet());
+
 #elif defined(__DragonFly__) || \
       defined(__OpenBSD__)   || \
       defined(__FreeBSD__)   || \
       defined(__NetBSD__)    || \
       defined(__APPLE__)
-    int mib[2];
+    ncpu = 0;
     size_t len = sizeof(ncpu);
-    mib[0] = CTL_HW;
-    mib[1] = HW_NCPU;
-    if (sysctl(mib, 2, &ncpu, &len, NULL, 0) != 0)
+    int mib[2] = {CTL_HW, HW_NCPU};
+    if (sysctl(mib, 2, &ncpu, &len, NULL, 0) != 0) {
         ncpu = 0;
+    }
 #endif
-    if (ncpu >= 1)
-        return PyLong_FromLong(ncpu);
-    else
+
+    if (ncpu < 1) {
         Py_RETURN_NONE;
+    }
+    return PyLong_FromLong(ncpu);
+}
+
+
+/*[clinic input]
+os.process_cpu_count
+
+Get the number of logical CPUs usable by the current process.
+
+Return None if indeterminable.
+[clinic start generated code]*/
+
+static PyObject *
+os_process_cpu_count_impl(PyObject *module)
```

Comment:
Is there any reason to implement this in C?  Can't this be implemented in Python just using `return min(os.cpu_count(), len(os.sched_getaffinity(0)))` with additional None or empty set edge case handling added?

Sure that temporary set may not be efficient, but this isn't a function that should need efficiency for its use cases.

Reply:
Hum. I started to write a rationale to justify that, but I instead, I reimplemented the function in pure Python: it's simpler :-)

---

My first version also had an implementation for Windows which requires C code. But we can reconsider C vs Python implementation when we will consider adding a Windows implementation.

Reply:
Windows requiring C bits suggests we should just add `sched_getaffinity` for Windows (as it's own feature request and PR).  https://learn.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getprocessaffinitymask (which I think you had used) supports up to 64, and the hierarchical complexity added by https://learn.microsoft.com/en-us/windows/win32/procthread/processor-groups appears to add support for a whole lot more.